content/en/docs/Customizing/stages.md

```diff
@@ -127,3 +127,174 @@ stages:
        - |
           touch /run/sentinel
 ```
+
+#### after-install
+
+This stage is executed after installation of the OS has ended (after calling `cos-install`).
+
+Example:
+```yaml
+name: "Run something after installation"
+stages:
+   after-install:
+     - name: "Setting"
+       if: '[ ! -f "/run/cos/recovery_mode" ]'
+       commands:
+       - |
+          # Run something when we are booting in active or passive
+     - name: "Setting"
+       if: '[ -f "/run/cos/recovery_mode" ]'
+       commands:
+       - |
+          # Run something when we are booting in recovery mode
+```
+
+
+#### after-upgrade
+
+This stage is executed after upgrade of the OS has ended (after calling `cos-upgrade`).
+
+Example:
+```yaml
+name: "Run something after upgrade"
+stages:
+   after-upgrade:
+     - name: "Setting"
+       if: '[ ! -f "/run/cos/recovery_mode" ]'
+       commands:
+       - |
+          # Run something when we are booting in active or passive
+     - name: "Setting"
+       if: '[ -f "/run/cos/recovery_mode" ]'
+       commands:
+       - |
+          # Run something when we are booting in recovery mode
+```
+
+#### after-reset
+
+This stage is executed after reset of the OS has ended (after calling `cos-reset`).
+
+Example:
+```yaml
+name: "Run something after upgrade"
```

Comment:
the names are the same in the examples. just a nit

Reply:
argh! :sweat_smile: will fix it :)